File: apps/api/src/scripts/read-excel.ts
import XLSX from 'xlsx';
import { join } from 'path';
import { existsSync } from 'fs';

/**
 * Script para ler a planilha Excel e mostrar sua estrutura
 * Execute com: pnpm read-excel
 */
function readExcel() {
  const filePath = join(process.cwd(), '../../planilha_financeira_completa.xlsx');
  
  try {
    if (!existsSync(filePath)) {
      console.error(`❌ Arquivo não encontrado: ${filePath}`);
      console.error('💡 Certifique-se de que o arquivo está na raiz do projeto.');
      process.exit(1);
    }

    console.log('📊 Lendo planilha:', filePath);
    console.log('');
    
    const workbook = XLSX.readFile(filePath);
    const result: any = {
      totalSheets: workbook.SheetNames.length,
      sheets: [],
      structure: {},
    };

    workbook.SheetNames.forEach((sheetName, index) => {
      console.log(`📄 Processando aba ${index + 1}/${workbook.SheetNames.length}: "${sheetName}"`);
      
      const worksheet = workbook.Sheets[sheetName];
      const jsonData = XLSX.utils.sheet_to_json(worksheet, { 
        header: 1, 
        defval: '',
        raw: false 
      }) as any[][];
      
      const rows = jsonData.slice(0, 20); // Limitar a 20 linhas para preview
      
      const sheetInfo = {
        name: sheetName,
        totalRows: jsonData.length,
        totalColumns: rows[0]?.length || 0,
        hasData: jsonData.length > 0,
      };
      
      result.sheets.push(sheetInfo);

      if (rows.length > 0) {
        result.structure[sheetName] = {
          headers: rows[0] || [],
          sampleData: rows.slice(1, 6) || [],
          totalRows: jsonData.length,
        };
        
        console.log(`   ✓ ${jsonData.length} linhas, ${rows[0]?.length || 0} colunas`);
        console.log(`   ✓ Cabeçalhos: ${(rows[0] || []).join(' | ')}`);
      } else {
        console.log(`   ⚠️  Aba vazia`);
      }
    });

    console.log('');
    console.log('📋 Estrutura Completa da Planilha:');
    console.log('='.repeat(60));
    console.log(JSON.stringify(result, null, 2));
    console.log('');
    console.log('💡 Use essas informações para criar o script de importação customizado.');
    
    return result;
  } catch (error: any) {
    console.error('❌ Erro ao ler planilha:', error.message);
    console.error('\n💡 Certifique-se de que:');
    console.error('   1. O arquivo existe na raiz do projeto');
    console.error('   2. A biblioteca xlsx está instalada: pnpm install');
    console.error('   3. O arquivo não está corrompido');
    process.exit(1);
  }
}

readExcel();

File: scripts/read_excel.py
#!/usr/bin/env python3
"""
Script temporário para ler a planilha Excel e mostrar sua estrutura
"""
import sys
import json

try:
    import openpyxl
    
    def read_excel(file_path):
        """Lê o arquivo Excel e retorna a estrutura"""
        wb = openpyxl.load_workbook(file_path)
        result = {
            "sheets": [],
            "structure": {}
        }
        
        for sheet_name in wb.sheetnames:
            sheet = wb[sheet_name]
            rows = []
            
            # Ler até 20 linhas para entender a estrutura
            for idx, row in enumerate(sheet.iter_rows(values_only=True), 1):
                if idx > 20:  # Limitar a 20 linhas por sheet
                    break
                if any(cell is not None for cell in row):  # Ignorar linhas vazias
                    rows.append([str(cell) if cell is not None else '' for cell in row])
            
            result["sheets"].append({
                "name": sheet_name,
                "rows": rows,
                "max_row": sheet.max_row,
                "max_column": sheet.max_column
            })
            
            # Primeira linha geralmente é o cabeçalho
            if rows:
                result["structure"][sheet_name] = {
                    "headers": rows[0] if len(rows) > 0 else [],
                    "sample_data": rows[1:6] if len(rows) > 1 else []
                }
        
        return result
    
    if __name__ == "__main__":
        file_path = "planilha_financeira_completa.xlsx"
        data = read_excel(file_path)
        print(json.dumps(data, indent=2, ensure_ascii=False))
        
except ImportError:
    print("ERRO: openpyxl não está instalado. Instale com: pip install openpyxl")
    sys.exit(1)
except Exception as e:
    print(f"ERRO ao ler arquivo: {e}")
    sys.exit(1)

